lib/src/logic.dart

```diff
@@ -424,8 +424,12 @@ class Logic {
   }
 
   /// Accesses the [index]th bit of this signal.
-  Logic operator [](int index) {
-    return slice(index, index);
+  Logic operator [](dynamic index) {
+      if (index is Logic) {
+          return IndexGate(this, index).y;
+      }
+
```

Comment:
It would be a good idea to explicitly check if `index` is neither an `int` nor a `Logic` and throw and `Exception` in that case explaining that it only accepts int or Logic.